### PR TITLE
Add ability to tag source repository and retrieve release version fro…

### LINKED ISSUE #809: Support maintaining repository tags for release builds
**Describe the solution you'd like**

We use cekit to build our container images. We publish container images from builds triggered by cekit. We would like to tag the container source repository to indicate which commits correspond to published images.

For example, if we built and shipped a container named foo which had a version: 1.0, I would like to tag the corresponding source repository commit with foo-1.0, or similar.

There are a bunch of issues here, but most fundamentally: cekit doesn't perform the release; that happens externally. So it's not as  simple as adding a tag to the `build --release` workflow. On the other hand, doing it manually (as now) opens up the risk of making mistakes, and once you've published a tag, you can't reasonably un-publish it (I've just found a mistake, which is what motivated me to finally file this enhancement request)

Perhaps cekit could create, but not push, tags in the local clone of the container sources, with a convention like "foo-1.0-candidate" or similar; leaving the user to create the real tag (`git tag foo-1.0 foo-1.0-candidate`) and push that. It would reduce the risk of mistakes at least.

### PR DESCRIPTION
**Pull request by rnc**
…m OSBS for unique tags

Fixes #809

### CODE REVIEW

**Comment by jmtd** [comment]
I'm really grateful for this feature, thanks! I've reviewed it and I've got a couple of comments but nothing blocking.

 * `--help` output description of `--tag` suggests that it takes an argument in all cases (`--tag TAG                 Use specified tag to tag the dist-git repository after build`), I don't think that's consistent with how it currently works
 * a scratchbuild (`--tag` without `--release`) won't work for tagging; cekit does not catch that at least in early stages of invocation (I'll return to this)

**Comment by rnc** [comment]
You are correct regarding the CLI help. I can add 
```
  +    default="{{name}}-{{version}}",
  +    show_default=True,
```
which will give
```
  --commit-message MESSAGE  Custom dist-git commit message.
  --tag TAG                 Use specified tag to tag the dist-git repository after build
                            [default: {{name}}-{{version}}]
```
which matches other CLI arguments that take an optional component with a known default. 

Regarding scratch builds - I thought I wrote it so it would **not** try to tag - the code should output:
```
            LOGGER.warning(
                f"No koji build found - maybe {task_id} was a scratch build?"
            )
```

**Comment by jmtd** [comment]
>  I can add

Looks perfect

> Unless - did you mean it doesn't try to catch it earlier ?

Yes, I meant earlier: I think now `--tag` doesn't make sense without also `--release`. In fact, is this a problem? Perhaps there's a use-case for tagging when not doing releases. Or even using OSBS but that's a bigger thing (since this is osbs subsystem specific code). Of course I'm speculating, because I am using it for OSBS release builds... 

I don't think this is a blocking issue fwiw!

**Comment by jmtd** [comment]
Example annotated tags composed by this PR: https://github.com/jboss-container-images/openjdk/releases/tag/openjdk-openjdk-11-rhel7-1.15-6 and https://github.com/jboss-container-images/openjdk/releases/tag/openjdk-openjdk-11-rhel7-1.15-8

**Review by jmtd — DISMISSED** [review]

File: cekit/builders/osbs.py [hunk 1]
```diff
@@ -340,15 +341,63 @@ def run(self):
...
+            # This is a bit of code reuse cheat - if tagging is enabled this will also tag
+            # the image source repository (while the above block handles the dist-git source)
+            url = run_wrapper(
+                ["git", "config", "--get", "remote.origin.url"],
+                capture_output=True,
+                check=False,
+            ).stdout
+            if git_tag:
+                self.git.tag(url, git_tag, build_id)
+                self.git.push(True)
```
[the remark this answers is not in the document]
Comment (rnc) [reply cekit/builders/osbs.py:358]: Yes it does. That is a good point, I'll change it to `git push origin <tag-name>`

**Comment by rnc** [comment]
> 
> Yes, I meant earlier: I think now `--tag` doesn't make sense without also `--release`. In fact, is this a problem? Perhaps there's a use-case for tagging when not doing releases. Or even using OSBS but that's a bigger thing (since this is osbs subsystem specific code). Of course I'm speculating, because I am using it for OSBS release builds...
> 
> I don't think this is a blocking issue fwiw!

I think we could do this with a callback function in click

File: cekit/builders/osbs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmtd) [reply cekit/builders/osbs.py]: Just retried, looks great, thank you! This was pushed: https://github.com/jboss-container-images/openjdk/releases/tag/redhat-openjdk-18-openjdk18-openshift-1.15-6

**Review by jmtd — APPROVED** [review]

**Pull request merged**